File: 2/JS/funcs.js
function fibonacci(limit) {
    var i = 0;
    var j = 1;
    result = [];
    while (i <= limit) {
        result.push(i);
        i = j;
        j = result[result.length - 1] + j;
    }
    return result;
}

function sum_even_fibonacci(limit) {
    var fibo = fibonacci(limit);
    result = [];
    for (i=0; i <= fibo.length - 1; i++) {
        if (fibo[i] % 2 === 0) {
            result.push(fibo[i]);
        }
    }
    return result.reduce((sum, value) => sum + value)
}

File: 2/R/funcs.R
fibonacci <- function(limit) {
    i = 0
    j = 1
    result = c()
    while(i <= limit) {
        result = append(result, i)
        i = j
        j = j + result[length(result)]
    }
    return(result)
}

sum_even_fibonacci <- function(limit) {
    fibo = fibonacci(limit)
    sum_fibo = 0
    for (num in fibo) {
        if (num %% 2 == 0) {
            sum_fibo = sum_fibo + num
        }
    }
    return(sum_fibo)
}
